### PR TITLE
Add an end tag at the end of the post

### PR DESCRIPTION
**Pull request by upcwangying**
## PR Checklist
**Please check if your PR fulfills the following requirements:**

- [X] The commit message follows [our guidelines](https://github.com/theme-next/hexo-theme-next/blob/master/.github/CONTRIBUTING.md).
- [X] Tests for the changes was maked (for bug fixes / features).
   - [X] Muse | Mist have been tested.
   - [X] Pisces | Gemini have been tested.
- [x] Docs in [NexT website](https://theme-next.org/docs/) have been added / updated (for new features).

## PR Type
**What kind of change does this PR introduce?**

- [ ] Bugfix.
- [X] Feature.
- [ ] Code style update (formatting, local variables).
- [ ] Refactoring (no functional changes, no api changes).
- [ ] Build related changes.
- [ ] CI related changes.
- [ ] Documentation content changes.
- [ ] Other... Please describe:

## What is the current behavior?

Add an end tag at the end of the post

Issue resolved: N/A

## What is the new behavior?

Add an end tag at the end of the post

- Screenshots with this changes: N/A
- Link to demo site with this changes: N/A

### How to use?
In NexT `_config.yml`:
```yml
post_end_tag:
  enable: true
  message: Thanks for reading
```

## Does this PR introduce a breaking change?
- [ ] Yes.
- [X] No.

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -1205,6 +1205,12 @@ vendors:
   disqusjs_js:
   disqusjs_css:
 
+# Add an end tag at the end of the post
```
Comment (stevenjoezhang) [inline _config.yml]: Thanks for your contributing. Would you like to move this to `Post Settings` section in `_config.yml`?

File: layout/_macro/post-end-tag.swig [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<div>
+  {% if theme.post_end_tag.enabled and theme.post_end_tag.message %}
+    <div style="text-align:center;color: #ccc;font-size:14px;">{{ theme.post_end_tag.message }}</div>
```
Comment (stevenjoezhang) [inline layout/_macro/post-end-tag.swig]: And remove the hard-coded `font-size` here. Let it inherit from the parent element. See: https://github.com/theme-next/hexo-theme-next/pull/800

**Comment by stevenjoezhang** [comment]
And the docs about this feature here: https://github.com/theme-next/theme-next.org/blob/source/source/docs/theme-settings/posts.md

File: _config.yml (hunk 1 above)
Comment (upcwangying) [reply _config.yml]: Where is `Post Settings` section? Is this location?
批注 2019-04-15 113537

Comment (stevenjoezhang) [reply _config.yml]: Yes, it's here:
https://github.com/theme-next/hexo-theme-next/blob/933dd90ae731561247d17b668273f8b3f7a8b869/_config.yml#L291-L389

Comment (stevenjoezhang) [reply _config.yml]: My suggestion is to put it in line 359, before the `wechat_subscriber` settings. How do you like it? Thanks.

File: _config.yml [hunk 3]
```diff
@@ -347,6 +347,12 @@ codeblock:
     # Style: only 'flat' is currently available, leave it blank if you prefer default theme
     style:
 
+# Add an end tag at the end of the post
+# if use, set `enabled` to true
+post_end_tag:
+  enabled: false
```
Comment (Raincal) [inline _config.yml]:
```suggestion
  enable: false
```

**Comment by upcwangying** [comment]
Docs updated

**Comment by ivan-nginx** [comment]
Maybe we must to refactor same feature for add custom fields anywhere. @JiangTJ seems suggest something like this.

**Comment by 1v9** [comment]
> Maybe we must to refactor same feature for add custom fields anywhere. @JiangTJ seems suggest something like this.

@ivan-nginx Like WordPress Widgets?

**Comment by jiangtj** [comment]
https://github.com/theme-next/hexo-theme-next/pull/868 Hi, guy. Can you review this, does it meet your needs?

My test site: https://jiangtj.gitlab.io/next-test/muse/post/hello-world/

Theme config:
```yml
custom_file_path:
  postBodyEnd: source/_data/post-body-end.swig
```

In `source/_data/post-body-end.swig`
```html
<div style="text-align: center; color: #ccc; font-size: 1em;">Thank you for reading</div>
```

**Comment by upcwangying** [comment]
@jiangtj Yes, it meets my needs.

**Comment by stevenjoezhang** [comment]
@upcwangying Thanks for your contributing. We will add this feature in #868

**Pull request closed without merging**